storage/google/cloud/storage/notification.py

```diff
@@ -129,6 +129,8 @@ def from_api_repr(cls, resource, bucket):
 
         name = match.group('name')
         project = match.group('project')
+        if project and len(project) > 30:
```

Comment:
@chemelnucfin The length is **part** of the regex, why are you checking it?

Reply:
I had a failing contrived test case:
'a' * 31, matched, but an invalid project id because it was too long.

Reply:
I also tried word boundaries and such, but did not work combining with the topics.